eng/bomgenerator/buildAndRun.cmd

```diff
@@ -0,0 +1,4 @@
+set "versioningClientFileLocation=%~dp0..\versioning\version_client.txt"
+set "bomPomFileLocation=%~dp0..\..\sdk\boms\azure-sdk-bom\pom.xml"
+set "outputFileLocation=%~dp0..\..\sdk\boms\azure-sdk-bom\newpom.xml"
+mvn clean install && mvn exec:java -Dexec.args="-inputFile=%versioningClientFileLocation% -outputFile=%outputFileLocation% -pomFile=%bomPomFileLocation%"
```

Comment:
Out of curiosity is there a reason you decided to write a java based tool for generating this? Not that it matters too much but most of the other versioning tools are written in python so this would need to duplicate a chunk of that logic. Do we think we should move more of the versioning tooling to java?

Reply:
There were three main reasons 1. I rely on a library shrinkwrap that basically emulates maven assembly resolution to create the dependency map for each library. I also depend on maven reader and writer library which is again available for java. (I believe there are solutions that can call java methods from python directly but I did not delve into that much) 2. This can in future become a plug-in and be part of a build step. 3. My vested interest in getting more comfy in Java. 

When you say duplication of code - do you mean parsing the files and versions etc, or other things too?

Reply:
> Do we think we should move more of the versioning tooling to java?
This is a tough one. I see why python is the language of choice here? But at the same time for scenarios like BOM having things in java and reuse library\solutions from the ecosystem is beneficial.

Reply:
Thanks for the context. I don't have an issue with this, especially if this becomes a plugin and produced as part of the maven build. 

The duplication I was thinking about is the parsing and understanding of our version files. It isn't overly complicated but there are some interesting rules in there to understand dependency vs current versions as well as our unreleased/beta versions.

One of the biggest reason we use things like python and powershell scripts for our versioning and other engsys tasks is to avoid needing to do a restore and build step before using the tool. Interpreted languages are much better suited for those type of tasks. For that reason it might be kind of difficult to implement all the versioning rules in a java based tool, at least one that lives in this repo. If we did use a java tool for these type of things I'd suggest we move the building/publishing of that tool into another repo.

Reply:
I think @srnagar also said having a plugin which lived in a repo like azure-sdk would be a better place to put this. How about I make this the first step and use it to polish the tool a little and once it is mature and ready to be turned into a plugin, I can move it a more suited location.

Reply:
> One of the biggest reason we use things like python and powershell scripts for our versioning and other engsys tasks is to avoid needing to do a restore and build step before using the tool. Interpreted languages are much better suited for those type of tasks.

I did not think of this .. thanks!

Reply:
If you do end up putting this in another repo the right place for it would be https://github.com/Azure/azure-sdk-tools/tree/main/tools.

Reply:
Thanks :)